packages/react-provider/src/hooks/use-client-source-id.ts

```diff
@@ -0,0 +1,37 @@
+import useHash from "./use-hash";
+import * as uuid from "uuid";
+import { useMemo } from "react";
+
+const useClientSourceId = ({
+  clientKey,
+  userId,
+  authorization,
+  id,
+  localStorage,
+}): string => {
+  const clientSourceKey = useHash(
+    authorization ? authorization : `${clientKey}/${userId}`
+  );
+
+  return useMemo(() => {
+    if (!localStorage) {
```

Comment:
If localstorage is required, why would we need this check?

Reply:
its not required